code/modules/scrap/scrap_pileripper.dm

```diff
@@ -0,0 +1,165 @@
+/obj/item/weapon/circuitboard/pile_ripper
+	name = T_BOARD("Pile Ripper")
+	build_path = /obj/machinery/pile_ripper
+	board_type = "machine"
+	origin_tech = list(TECH_ENGINEERING = 3)
+	req_components = list(
+		/obj/item/weapon/stock_parts/manipulator = 1
+	)
+
+/obj/machinery/pile_ripper
+	name = "pile ripper"
+	desc = "This machine rips everything in front of it apart."
+	icon = 'icons/obj/structures/scrap/recycling.dmi'
+	icon_state = "grinder-b0"
+	layer = MOB_LAYER + 1 // Overhead
+	anchored = TRUE
+	density = TRUE
+	use_power = 1
+	idle_power_usage = 300
+
+	var/safety_mode = FALSE // Temporality stops the machine if it detects a mob
+	var/icon_name = "grinder-b"
+	var/blood = 0
+	var/rating = 1
+	var/last_ripped = 0
+
+	var/turf/ripped_turf
+
+/obj/machinery/pile_ripper/Initialize()
+	..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/pile_ripper/LateInitialize()
+	ripped_turf = get_turf(get_step(src, 8))
+
+/obj/machinery/pile_ripper/Process()
+	if(last_ripped >= world.time)
+		return
+	last_ripped = world.time
+	if(safety_mode)
+		playsound(loc, 'sound/machines/ping.ogg', 50, 0)
+		safety_mode = FALSE
+		update_icon()
+	for(var/mob/living/poor_soul in ripped_turf)
+		if(emagged || prob(25))
+			eat(poor_soul)
+		else
+			stop(poor_soul)
+	var/count = 0
+	for(var/obj/ripped_item in ripped_turf)
+		if(count >= rating)
+			break
+		if(istype(ripped_item, /obj/structure/scrap))
+			var/obj/structure/scrap/pile = ripped_item
+			while(pile.dig_out_lump(loc, 1))
+				if(prob(20))
+					break
+			count++
+		else if(istype(ripped_item, /obj/item))
+			ripped_item.forceMove(src.loc)
+			if(prob(20))
+				qdel(ripped_item)
+		else if(istype(ripped_item, /obj/structure/scrap_cube))
+			var/obj/structure/scrap_cube/cube = ripped_item
+			cube.make_pile()
+
+/obj/machinery/pile_ripper/examine(mob/user)
+	..()
+	to_chat(user, "The power light is [(stat & NOPOWER) ? "off" : "on"].")
+	to_chat(user, "The safety-mode light is [safety_mode ? "on" : "off"].")
+	to_chat(user, "The safety-sensors status light is [emagged ? "off" : "on"].")
+
+/obj/machinery/pile_ripper/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/pile_ripper/proc/stop(mob/living/L)
+	playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
+	safety_mode = TRUE
+	update_icon()
+	L.forceMove(loc)
+
+	last_ripped += SAFETY_COOLDOWN
+	update_icon()
+
+/obj/machinery/pile_ripper/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/card/emag))
+		emag_act(user)
+		user.setClickCooldown(DEFAULT_QUICK_COOLDOWN)
+
+	if(default_deconstruction(I, user))
+		return
+
+	if(default_part_replacement(I, user))
+		return
+
+	..()
+
+/obj/machinery/pile_ripper/emag_act(mob/user)
+	if(!emagged)
+		emagged = TRUE
+		if(safety_mode)
+			safety_mode = FALSE
+			update_icon()
+		playsound(loc, "sparks", 75, 1, -1)
```

Comment:
pretty sure this should be src rather than loc but I'm not entirely sure if playsound here works like /tg/'s which does recommend people use src rather than its loc